README.md

```diff
@@ -102,23 +105,99 @@ Environment:
-mosquitto_pub -u <thing_id> -P <thing_key> -t channels/<control_channel>/messages/services/adc -h <mqtt_host> -p 1883  -m  "[{\"bn\":\"1:\", \"n\":\"read\", \"vs\":\"temperature\"}]"
+
+## Heartbeat service
+Services running on the same host can publish to `heartbeat.<service-name>.<service-type>` a heartbeat message.  
+Agent will keep a record on those service and update their `live` status.
+If heartbeat is not received in 10 sec it marks it `offline`.
+Upon next heartbeat service will be marked `online` again.
+
+To check services that are currently registered to agent you can:
+
+```bash
+curl -s -S X GET http://localhost:9000/services
 ```
+```json
+[
+  {
+    "name": "duster",
+    "last_seen": "2020-04-28T18:06:56.158130519+02:00",
+    "status": "offline",
+    "type": "test",
+    "terminal": 0
+  },
+  {
+    "name": "scrape",
+    "last_seen": "2020-04-28T18:06:39.58849766+02:00",
+    "status": "offline",
+    "type": "test",
+    "terminal": 0
+  }
+]
+```
+
+Or you can send a command via MQTT to Agent and receive response on MQTT topic like this:
 
+In one terminal subscribe for result:
+
+```bash
+mosquitto_sub -u <thing_id> -P <thing_key> -t channels/<control_channel_id>/messages/req -h <mqtt_host> -p 1883  
+```
+
+In another terminal publish request to view the list of services:
+
+```bash
+mosquitto_pub -u <thing_id> -P <thing_key> -t channels/<control_channel_id>/messages/req -h <mqtt_host> -p 1883  -m  '[{"bn":"1:", "n":"config", "vs":"view"}]'  
+```
+
+Check the output in terminal where you subscribed for results. You should see something like:
+
+```json
+[
+  {
+    "bn": "1",
+    "n": "view",
+    "t": 1588091188.8872917,
+    "vs": "[{\"name\":\"duster\",\"last_seen\":\"2020-04-28T18:06:56.158130519+02:00\",\"status\":\"offline\",\"type\":\"test\",\"terminal\":0},{\"name\":\"scrape\",\"last_seen\":\"2020-04-28T18:06:39.58849766+02:00\",\"status\":\"offline\",\"type\":\"test\",\"terminal\":0}]"
+  }
+]
+```
+
+
+## How to save config via agent
+Agent can be used to send configuration file for the [Export][export] service from cloud to gateway via MQTT.
+Here is the example command:
+
+```bash
+mosquitto_pub -u <thing_id> -P <thing_key> -t channels/<control_ch_id>/messages/req -h localhost -p 18831  -m  "[{\"bn\":\"1:\", \"n\":\"config\", \"vs\":\"config.toml, RmlsZSA9ICIuLi9jb25maWdzL2NvbmZpZy50b21sIgoKW2V4cF0KICBsb2dfbGV2ZWwgPSAiZGVidWciCiAgbmF0cyA9ICJuYXRzOi8vMTI3LjAuMC4xOjQyMjIiCiAgcG9ydCA9ICI4MTcwIgoKW21xdHRdCiAgY2FfcGF0aCA9ICJjYS5jcnQiCiAgY2VydF9wYXRoID0gInRoaW5nLmNydCIKICBjaGFubmVsID0gIiIKICBob3N0ID0gInRjcDovL2xvY2FsaG9zdDoxODgzIgogIG10bHMgPSBmYWxzZQogIHBhc3N3b3JkID0gImFjNmI1N2UwLTliNzAtNDVkNi05NGM4LWU2N2FjOTA4NjE2NSIKICBwcml2X2tleV9wYXRoID0gInRoaW5nLmtleSIKICBxb3MgPSAwCiAgcmV0YWluID0gZmFsc2UKICBza2lwX3Rsc192ZXIgPSBmYWxzZQogIHVzZXJuYW1lID0gIjRhNDM3ZjQ2LWRhN2ItNDQ2OS05NmI3LWJlNzU0YjVlOGQzNiIKCltbcm91dGVzXV0KICBtcXR0X3RvcGljID0gIjRjNjZhNzg1LTE5MDAtNDg0NC04Y2FhLTU2ZmI4Y2ZkNjFlYiIKICBuYXRzX3RvcGljID0gIioiCg==\"}]"
```

Comment:
We do not fullblown cert string here, replace by `<cert>`

Reply:
Explain also which kind of cert this is - CA, pub or priv X.509. For example `<ca_cert>` or `<pub_cert_x509>` or something...